File: 202 Happy Number/isHappyNumber.cpp
class Solution {
public:
    bool isHappy(int n) {
        
        if( n == 1 )
                return true;
        
        unordered_set<int> sumResult;
        
        int current = n;
        while( current > 0 ){
            
            int sum = 0;
            while( current > 0){
                sum += (current % 10) * (current % 10);
                current /= 10;
            }
            if(sum == 1)
                    return true;
            current = sum;
            
            if(sumResult.find(sum) != sumResult.end())
                return false;
            else
                sumResult.insert(sum);
        }
        return false;
    }
};

File: 202 Happy Number/isHappyNumber.py
class Solution(object):
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        if n == 1:
            return True;
        
        sumResult = set();
        current = n
        while current > 0:
            
            temp_sum = 0
            while current > 0:
                temp_sum += (current % 10) * (current % 10)
                current /= 10
            
            if temp_sum == 1:
                return True
            current = temp_sum
            
            if temp_sum in sumResult:
                return False
            else:
                sumResult.add(temp_sum)
            
        return False